pinot-core/src/main/java/org/apache/pinot/core/plan/CombinePlanNode.java

```diff
@@ -172,7 +173,19 @@ public CombinePlanNode(List<PlanNode> planNodes, QueryContext queryContext, Exec
       return new StreamingSelectionOnlyCombineOperator(operators, _queryContext, _executorService, _endTimeMs,
           _streamObserver);
     }
-    if (QueryContextUtils.isAggregationQuery(_queryContext)) {
+    if (QueryContextUtils.isSelectionQuery(_queryContext)) {
```

Comment:
Common case (of aggr/group-by) should be first?

Reply:
Done